File: leetcode/C++/longest-uncommon-subsequence-ii.cpp
class Solution {
public:
    bool isSubsequence(string &a, string &b){
        int idx=0;
        for(int i=0; i<b.length()&&idx<a.length(); ++i) if(a[idx]==b[i]) ++idx;
        return idx==a.length();
    }
    int findLUSlength(vector<string>& strs) {
        int sol=-1;
        for(int i=0,j=0; i<strs.size(); ++i){
            for(j=0; j<strs.size(); ++j){
                if(i!=j&&isSubsequence(strs[i],strs[j])) break;
            }
            if(j==strs.size()) sol=max(sol,(int)strs[i].length());
        }
        return sol;
    }
};

File: leetcode/Python/longest-uncommon-subsequence-ii.py
class Solution:
    def isSubsequence(self, a, b):
        idx=0
        for i in b:
            if a[idx]==i:
                idx+=1
            if idx>=len(a):
                return True
        return False
    def findLUSlength(self, strs: List[str]) -> int:
        sol = -1
        for i in range(len(strs)):
            check=True
            for j in range(len(strs)):
                if i!=j and self.isSubsequence(strs[i],strs[j]):
                    check=False
                    break
            if check:
                sol = max(sol, len(strs[i]))
        return sol